frontend/lib/swr-cache/index.ts

```diff
@@ -35,6 +36,7 @@ interface CacheOptions<
    ValueSchema extends z.ZodTypeAny
 > {
    endpoint: string;
+   statName: string;
```

Comment:
> This is a point of friction when defining and endpoint, where the developer has to understand the purpose of this, and has to come up with a name.

Coming up with a human readable name for the thing we are trying to cache doesn't seem like a point of friction. If this ended up being used hundreds of times... then *maybe*, but it'll probably be a dozen or so. 

Without a human readable name, we are co-opting an implementation detail (the endpoint path) to substitute for a human readable name in the logs.

> When I was building this abstraction I also looked for a way to get rid of the endpoint option

I think both are important enough to have it a required parameter, rather than a magic default.

Regarding `statName`, you're right that it's worth documenting somewhere.

Reply:
> If this ended up being used hundreds of times... then maybe, but it'll probably be a dozen or so.

I understand your argument, however, I believe we should consider these choices in the context of the entire project. Every little detail compounds over time.

> Without a human readable name, we are co-opting an implementation detail (the endpoint path) to substitute for a human readable name in the logs.

Maybe it's just that I don't see `api/nextjs/cache/product` that different from `cache.findProduct`. Maybe I'm wrong, but I think the purpose of this label is to help you identify what a given log is referring to, and I'm not sure `cache.findProduct` would be more helpful than `api/nextjs/cache/product`. 
I don't have enough context over what you're doing with Datadog so if you think it's important to keep this name than I'm fine with it 👍